### PR TITLE
Run modernize over the code

### PR DESCRIPTION
**Pull request by jgraham**
This provides initial work on compatibility with Python 3

### CODE REVIEW

**Comment by karlcow** [comment]
@jgraham do you need to keep the python 2 compatibility?

**Comment by jgraham** [comment]
Yes, it needs to keep working in Python 2; we are some way away from actually supporting 3.

File: sync/gh.py [hunk 1]
```diff
@@ -97,7 +99,7 @@ def remove_labels(self, pr_id, *labels):
                     newrelic.agent.record_exception()
 
     def _convert_pr_id(self, pr_id):
-        if not isinstance(pr_id, (int, long)):
+        if not isinstance(pr_id, six.integer_types):
```
Comment (karlcow) [inline sync/gh.py:102]: same comment than earlier about

```
from builtins import bytes, str
```

after future import

File: sync/landing.py [hunk 2]
```diff
@@ -1038,14 +1039,19 @@ def update_landing(git_gecko, git_wpt, prev_wpt_head=None, new_wpt_head=None,
...
     return landing
 
 
+def retrigger():
+    try:
+        # Avoid circular import
+        from . import tasks
+        tasks.retrigger.apply_async()
+    except OperationalError:
+        logger.warning("Failed to retrigger blocked syncs")
```
Comment (karlcow) [inline sync/landing.py:1052]: This seems to be completely unrelated to the conversion. Or maybe because of the import?

File: sync/listen.py [hunk 3]
```diff
@@ -188,13 +190,14 @@ def accept(self, body):
...
-    event_filters = {item: lambda x: True for item in handlers.GitHubHandler.dispatch_event.keys()}
+    event_filters = {item: lambda x: True
+                     for item in handlers.GitHubHandler.dispatch_event.keys()}
     event_filters["status"] = lambda x: x["payload"]["context"] != "upstream/gecko"
     event_filters["push"] = lambda x: x["payload"]["ref"] == "refs/heads/master"
 
     def __init__(self, config, logger):
         super(GitHubFilter, self).__init__(config, logger)
-        repo_path = urlparse.urlparse(config["web-platform-tests"]["repo"]["url"]).path
+        repo_path = urllib.parse.urlparse(config["web-platform-tests"]["repo"]["url"]).path
```
Comment (karlcow) [inline sync/listen.py:200]: This looks a better way to do it than before. In other parts of the code, the import to `six` spray `six.` keywords into the code instead of the ready to go python 3 value.

It could be a reminder to remove them when full python 3 conversion is reached or like here just the import line to change at the top.

File: sync/landing.py (hunk 2 above)
Comment (jgraham) [reply sync/landing.py:1052]: Yeah, we need to avoid the circular import, and it means we need to use a different approach to mocking this in the testing, so this is required to make the tests pass again.

File: sync/commit.py [hunk 4]
```diff
@@ -92,7 +94,7 @@ def __init__(self, repo, commit):
         elif hasattr(commit, "sha1"):
             # Commit subclass
             sha1 = commit.sha1
-        elif isinstance(commit, (str, unicode)):
+        elif isinstance(commit, (str, six.text_type)):
```
[the remark this answers is not in the document]
Comment (jgraham) [reply sync/commit.py]: I think this is just bad conversion; in Py2 we are accepting both here so I think it makes sense to still accept both until we figure out what the actual types are here.

File: sync/gh.py (hunk 1 above)
Comment (jgraham) [reply sync/gh.py:104]: This seems OK as is to me.

File: sync/base.py [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
-import log
-import commit as sync_commit
-from env import Environment
-from lock import MutGuard, RepoLock, mut, constructor
-from repos import pygit2_get
+from . import log
+from . import commit as sync_commit
+from .env import Environment
+from .lock import MutGuard, RepoLock, mut, constructor
+from .repos import pygit2_get
```
[the remark this answers is not in the document]
Comment (jgraham) [reply sync/base.py:15]: The `from __future__ import absolute_imports` forces you into the Python 3 import syntax, even though as you say the cleaned up imports are relative. I think this could be something to clean up in the future rather than in in this PR.

File: sync/commit.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (karlcow) [reply sync/commit.py]: yes the issue I was raising was not about having both types :) it was about the form and the logical order.
